nav2_graceful_motion_controller/CMakeLists.txt

```diff
@@ -0,0 +1,78 @@
+cmake_minimum_required(VERSION 3.5)
+project(nav2_graceful_motion_controller)
+
+find_package(ament_cmake REQUIRED)
+find_package(nav2_common REQUIRED)
+find_package(nav2_core REQUIRED)
+find_package(nav2_costmap_2d REQUIRED)
+find_package(nav2_util REQUIRED)
+find_package(rclcpp REQUIRED)
+find_package(geometry_msgs REQUIRED)
+find_package(nav_msgs REQUIRED)
+find_package(pluginlib REQUIRED)
+find_package(tf2 REQUIRED)
+find_package(tf2_geometry_msgs REQUIRED)
+find_package(nav_2d_utils REQUIRED)
+find_package(angles REQUIRED)
+
+nav2_package()
+set(CMAKE_CXX_STANDARD 17)
+
+include_directories(
+  include
+)
+
+set(dependencies
+  rclcpp
+  geometry_msgs
+  nav2_costmap_2d
+  pluginlib
+  nav_msgs
+  nav2_util
+  nav2_core
+  tf2
+  tf2_geometry_msgs
+  nav_2d_utils
+  angles
+)
+
+set(library_name nav2_graceful_motion_controller)
+
+add_library(${library_name} SHARED
```

Comment:
for ease of use with docking, might want to have two libraries - one that is just the underlying control law, and a second that is the actual controller plugin itself

Reply:
True! Or the main control law being header-only